src/main/scala/is/hail/utils/richUtils/RichHadoopConfiguration.scala

```diff
@@ -29,7 +29,7 @@ class RichHadoopConfiguration(val hConf: hadoop.conf.Configuration) extends AnyV
       os
   }
 
-  private def open(filename: String): InputStream = {
+  def open(filename: String): InputStream = {
```

Comment:
Gotta get us one of them fancy linear type systems.

[scala-arm](https://github.com/jsuereth/scala-arm) implements for-syntax for a `using`-like method. I looked at the code, it's rather complex, but it aims to handle the situation correctly in a variety of corner cases.

If you prefer a method per-resource, I will add whatever methods are necessary for this PR.

As an aside, my personal philosophy on the subject is that closing resources in Java/Scala is an unfortunately flawed part of the language/vm. We can reduce boilerplate with something like `using` or `scala-arm`, but ultimately PR review is the only way we can ensure that all resources are closed appropriately.